llvm/include/llvm/CodeGen/TargetLowering.h

```diff
@@ -5626,17 +5626,31 @@ class LLVM_ABI TargetLowering : public TargetLoweringBase {
   /// Get a pointer to vector element \p Idx located in memory for a vector of
   /// type \p VecVT starting at a base address of \p VecPtr. If \p Idx is out of
   /// bounds the returned pointer is unspecified, but will be within the vector
-  /// bounds.
-  SDValue getVectorElementPointer(SelectionDAG &DAG, SDValue VecPtr, EVT VecVT,
-                                  SDValue Index) const;
+  /// bounds. \p PtrArithFlags can be used to mark that arithmetic within the
+  /// vector in memory is known to not wrap or to be inbounds.
+  SDValue getVectorElementPointer(
+      SelectionDAG &DAG, SDValue VecPtr, EVT VecVT, SDValue Index,
+      const SDNodeFlags PtrArithFlags = SDNodeFlags()) const;
+
+  /// Get a pointer to vector element \p Idx located in memory for a vector of
+  /// type \p VecVT starting at a base address of \p VecPtr. If \p Idx is out of
+  /// bounds the returned pointer is unspecified, but will be within the vector
+  /// bounds. \p VecPtr is guaranteed to point to the beginning of a memory
+  /// location large enough for the vector.
+  SDValue getInboundsVectorElementPointer(SelectionDAG &DAG, SDValue VecPtr,
+                                          EVT VecVT, SDValue Index) const;
```

Comment:
```suggestion
  SDValue getInboundsVectorElementPointer(SelectionDAG &DAG, SDValue VecPtr,
                                          EVT VecVT, SDValue Index) const {
   return getVectorElementPointer(DAG, VecPtr, VecVT, Index,
     SDNodeFlags::NoUnsignedWrap | SDNodeFlags::InBounds);
}
```

Is there any compile-time advantage in having this simple function defined out-of-line?

Reply:
Probably not, you're right. I went for consistency with the surrounding declarations, but, looking further around in the file, there is precedent for inline definitions as well. Changed.